packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3489,6 +3504,25 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     return chain.join(' \u2190 ');
   }
 
+  /// Returns the first user created widget ancestor.
+  ///
+  /// Only available in debug builds with flag `--track-widget-creation`.
+  ///
+  /// Useful for debugging the source of an element.
+  String debugGetFirstLocalWidget() {
+    if (!WidgetInspectorService.instance.isWidgetCreationTracked()) {
+      return 'Enable widget creation tracking by using flutter run flag `--track-widget-creation`.';
+    }
+    Element node = this;
+    while (node != null && !isUserCreated(node)) {
+      node = node._parent;
+    }
+
+    return node != null ?
+      '${node.widget.toStringShort()} ${getCreationLocation(node)}' :
```

Comment:
All these entries can just be ErrorDescription('the string literal you already have')

Reply:
will you rather have
```ErrorDescription("First user created widget: \"Row file:///Users/chtai/apps/exception/lib/main.dart:16:22\"")```
or 
```
StringProperty("First user created widget", "Row file:///Users/chtai/apps/exception/lib/main.dart:16:22")
```
I currently choose StringProperty